### PR TITLE
feat(agent/grpc): add gRPC server

### PR DESCRIPTION
**Pull request by mmoreiradj**
## What am I looking at ???

Add a gRPC server that will run and stream the lifecycle of the agent to a caller. For now, it only streams the data once it's done.

## What still needs to be done

- [ ] Stream the data continuously during the build and the run
- [ ] Better error handling and logs
- [ ] Graceful shutdown of the running workload

(In another PR.)

## How can I run it ?

```bash
git git@github.com:mmoreiradj/cloudlet.git
cd cloudlet
git checkout feat/agent_grpc_server
```

Then the key `build.code-path` at `src/agent/examples/config.toml` to put the right path to the code.

```bash
cargo run --bin agent -- --config=./src/agent/examples/config.toml
```

> [!TIP]
> To use the debug agent you can use the feature flag `debug-agent` like this:
> ```bash
> cargo run --bin agent -F debug-agent -- --config=./src/agent/examples/config.toml
> ```

### CODE REVIEW

File: src/agent/src/workload/runner.rs [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
-        let agent: Box<dyn Agent> = match config.language {
+        let agent: Box<dyn Agent + Sync + Send> = match config.language {
             Language::Rust => Box::new(rust::RustAgent::from(config.clone())),
         };
 
         Runner { config, agent }
     }
 
-    pub fn run(&self) -> AgentResult<()> {
+    pub fn run(&self) -> AgentResult<AgentOutput> {
```
Comment (mfernd) [inline src/agent/src/workload/runner.rs:30]: You only return the logs of the `prepare`, `run` and run logs of `prepare-and-run`?
In `prepare-and-run`, you don't return the build logs

Comment (mfernd) [reply src/agent/src/workload/runner.rs:30]: I think we could do it later, it works like this

Comment (mmoreiradj) [reply src/agent/src/workload/runner.rs:30]: The goal is to stream it.

**Review by mfernd — APPROVED** [review]
I don't think the proto makes much sense, but for now they're fine...

**Comment by mmoreiradj** [comment]
Merged so the VMM team can work on communication with the agent.

**Pull request merged**